### DISCUSSION TITLE
Conditionally Execute SQL Query

### DISCUSSION THREAD

**Discussion by 711pranjal** [Budibase/budibase#8089] [Help]
Hi Team,

We are stuck in a situation where are trying to INSERT/DELETE row to/from a table using "On Change" functionality of checkbox.

With respect to condition:
1. If checkbox is True - execute insert SQL query.
2. If checkbox is False - execute update SQL query.

"Insert a row with status True, if user check the checkbox.
delete a row, if user un-check the checkbox"

Note 1 : We are trying to achieve this functionality dynamically(without using form Submit button).

Note 2: We even tried using automation to handle condition part but is with "AND" clause between multiple conditions.

<img width="720" alt="aspirations_table" src="https://user-images.githubusercontent.com/41870509/193521984-8d0fd066-121d-44fd-bcda-c06fe94412d7.PNG">
<img width="710" alt="select_user_aspiration_design" src="https://user-images.githubusercontent.com/41870509/193522465-1bc40bf3-c86e-4782-ac20-02b6485753ba.PNG">
<img width="955" alt="snap_career_portal" src="https://user-images.githubusercontent.com/41870509/193523085-0d94c203-1870-4515-b2c2-5b4d3216a79d.PNG">

**Comment by andz-bb** [comment]
hey @pranjalgupta711 

would something like this work for you? 

<img width="1293" alt="image" src="https://user-images.githubusercontent.com/110921612/193563355-d1631239-3f8a-4bf3-9f12-a836c12fca7b.png">

It won't be the most efficient because you are always running the delete first but it might allow you to do what you want to do within the constraints of the builder.

**Reply by melohagan** [reply]
I wonder if a CASE statement in the query would work as well?

**Comment by agruenberg** [comment]
I had a similar situation and solved it by a stored procedure call. Wherein I launched a insert into on duplicate key update query.
You could outsource the logic to the database.
